### ISSUE TITLE
Do Not Like Smaller Worker

### ISSUE THREAD

**Issue by SEATURTLEKING** [ScreepsOCS/screeps.behaviour-action-pattern#888]
I do not think the smaller worker when there no construction sites idea is very good because the workers still need to repair and give the tower energy when you are getting attacked, a worker with only a CARRY WORK and MOVE will not get the job done.

**Comment by Dasku** [comment]
++

I also think we have a bug with fortifying - none of my walls or ramparts are over 100k hits at RCL 8 - workers should be fortifying their hearts out

**Comment by karlthepagan** [comment]
What is the worker name?

**Comment by torin555** [comment]
I have these workers too. Things like worker-W78S38-1

**Comment by karlthepagan** [comment]
These are emergency room collapse workers. We need someone to implement the logic to ignore these when checking maxWorkers for a room. Then they should get recycled after the room is no longer collapsed.

**Comment by karlthepagan** [comment]
Related https://github.com/ScreepsOCS/screeps.behaviour-action-pattern/issues/525 https://github.com/ScreepsOCS/screeps.behaviour-action-pattern/issues/614

**Comment by SEATURTLEKING** [comment]
my room is not collapsed and they still spawn

**Comment by SEATURTLEKING** [comment]
the creep does not have to only have WORK CARRY MOVE
it can be a bit less powerful so it is actually useful

**Comment by karlthepagan** [comment]
If it costs more than 300 energy a room economy crash is still possible. This is a safety mechanism to prevent a room from being non-functional until the user intervenes.

**Comment by SEATURTLEKING** [comment]
yes but a creep with a body of WORK CARRY and MOVE cost 200, there is still 100 left over,
maybe if the user has another room the other room can help build extensions so you can make it so the creeps body is WORK, WORK, WORK, CARRY, CARRY, CARRY, MOVE, MOVE which cost 600 energy and is actually useful, 
also sometimes these 'recovery' creeps spawn when i am doing just fine it usually happens when i have no construction sites.

**Comment by SEATURTLEKING** [comment]
![e0c6e07d-ecf8-4909-9303-602fb4e80973](https://cloud.githubusercontent.com/assets/26051177/23491185/87d4317a-feb1-11e6-88cc-1468a6f07ae3.png)

**Comment by SEATURTLEKING** [comment]
![screenshot 2017-03-01 at 7 01 28 pm](https://cloud.githubusercontent.com/assets/26051177/23514510/6a164ba8-ff1c-11e6-8b3a-0b555dad7fa4.png)

**Comment by SEATURTLEKING** [comment]
worker top left corner

**Comment by karlthepagan** [comment]
do you have a screenshot or replay from when that worker spawned?

**Comment by SEATURTLEKING** [comment]
no this is the only thing i have

**Comment by chobobobo** [comment]
My working theory is that if your hauler limit for the room is 1 and no workers are needed, the room thinks it has collapsed during the spawn of the replacement hauler if the original hauler dies beforehand spawn completes.

**Comment by SEATURTLEKING** [comment]
ya that makes sense

**Comment by SEATURTLEKING** [comment]
one of my rooms got the recovery worker but the room has 2 energies

**Comment by karlthepagan** [comment]
please merge in #915 and debug that

I've got a lot going on but that should get you started

**Comment by xyzzy529** [comment]
The name of that worker "worker-E68S41-1" ... how did it get that name?
If it is a remoteWorker && the flag name is "E68S41" then it would be "remoteWorker-E68S41-1"
if it is a home worker then it is "worker-800-1" style.
I name my flags by the room and type. Are you doing that as well?

@SEATURTLEKING, is that how you name the flags?

**Comment by karlthepagan** [comment]
This is a function of the roomCollapse event which is handled by a task.

**Comment by karlthepagan** [comment]
Please let us know if the current Dev branch has resolved this issue.
